app/models/manageiq/providers/openstack/network_manager.rb

```diff
@@ -94,108 +94,31 @@ def self.event_monitor_class
-      :userid => userid
-    }
-    queue_opts = {
-      :class_name  => self.class.name,
-      :method_name => 'create_cloud_subnet',
-      :instance_id => id,
-      :priority    => MiqQueue::HIGH_PRIORITY,
-      :role        => 'ems_operations',
-      :zone        => my_zone,
-      :args        => [options]
-    }
-    MiqTask.generic_action_with_callback(task_opts, queue_opts)
-  end
-
-  def create_network_router(options)
-    NetworkRouter.raw_create_network_router(self, options)
-  end
-
-  def create_network_router_queue(userid, options = {})
-    task_opts = {
-      :action => "creating Network Router for user #{userid}",
-      :userid => userid
-    }
-    queue_opts = {
-      :class_name  => self.class.name,
-      :method_name => 'create_network_router',
-      :instance_id => id,
-      :priority    => MiqQueue::HIGH_PRIORITY,
-      :role        => 'ems_operations',
-      :zone        => my_zone,
-      :args        => [options]
-    }
-    MiqTask.generic_action_with_callback(task_opts, queue_opts)
-  end
-
-  def create_floating_ip(options)
-    FloatingIp.raw_create_floating_ip(self, options)
-  end
-
-  def create_floating_ip_queue(userid, options = {})
-    task_opts = {
-      :action => "creating Floating IP for user #{userid}",
-      :userid => userid
-    }
-    queue_opts = {
-      :class_name  => self.class.name,
-      :method_name => 'create_floating_ip',
-      :instance_id => id,
-      :priority    => MiqQueue::HIGH_PRIORITY,
-      :role        => 'ems_operations',
-      :zone        => my_zone,
-      :args        => [options]
-    }
-    MiqTask.generic_action_with_callback(task_opts, queue_opts)
-  end
-
-  def create_security_group(options)
-    SecurityGroup.raw_create_security_group(self, options)
-  end
-
-  def create_security_group_queue(userid, options = {})
-    task_opts = {
-      :action => "creating Security Group for user #{userid}",
-      :userid => userid
-    }
-    queue_opts = {
-      :class_name  => self.class.name,
-      :method_name => 'create_security_group',
-      :instance_id => id,
-      :priority    => MiqQueue::HIGH_PRIORITY,
-      :role        => 'ems_operations',
-      :zone        => my_zone,
-      :args        => [options]
-    }
-    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  # Builds create_<network item>_queue and associated raw_create_<network item> methods
+  [:CloudNetwork, :CloudSubnet, :FloatingIp, :NetworkRouter, :SecurityGroup].each do |subclass|
+    name = /([A-Z].*)([A-Z].*)/.match(subclass.to_s)
+    method_name = "#{name[1].downcase}_#{name[2].downcase}"
+    klass = Object.const_get("#{self}::#{subclass}")
+
+    define_method("create_#{method_name}") do |options|
+      klass.send("raw_create_#{method_name}".to_sym, self, options)
+    end
+
+    define_method("create_#{method_name}_queue") do |userid, options = {}|
+      task_opts = {
+        :action => "creating #{ui_lookup(:model => "subclass")} for user #{userid}",
```

Comment:
I think this is how `ui_lookup` works:

```
ui_lookup(:model => SomeActualClass.name)
```

or in your case `subclass` which is a symbol. Maybe it works with the symbol? Maybe you have to `.to_s` it.